### ISSUE TITLE
Naming question

### ISSUE THREAD

**Issue by jonaskello** [tizmagik/react-csjs#35]
For the most part the object that contains the styles is named `styles` and methods that operate on it is also named things like `withStyles`. But the corresponding prop is named `classes`. This seems a bit inconsistent to me so I just wanted to check if this was done for a reason?

The reason I ask is because I currently have my components using css modules and they accept a `styles` prop in order to override and that seems like the most used name for this prop. I want to switch to this lib but don't want to rename all props from styles to classes :-).

**Comment by tizmagik** [comment]
@jonaskello thanks for bringing this up. I'm open to suggestions here, but the reason we call the prop `classes` is because they are to be applied on the `className` prop, not the `style` prop. The value you get is not the style definition itself (e.g. what is returned from `csjs`) rather the generated _className_ that needs to be applied to the `className` prop in the component.

With that being said, you could rename it to whatever you'd like, e.g.

``` js
const {classes: styles} = this.props; // now available as `styles` instead of `classes`
```

I hope that answers your question. I'm open to other thoughts and suggestions!

**Comment by jonaskello** [comment]
Thanks for the explanation! I'm aware it is only the classname you get and I originally called my prop `classNames` for the reason you mention to not confuse with the style attribute. But I changed to `styles` since every css-modules example I encountered used this name. Eg. from the [spec repo](https://github.com/css-modules/css-modules)

``` js
import styles from "./style.css";
element.innerHTML = '<div class="' + styles.className + '">';
```

Regardless of which name is selected I do think the naming within a lib should be consistent. So if the prop should be called `classes` maybe the HOC should be called `withClasses`?

**Comment by tizmagik** [comment]
Actually the good news there is that the HoC is the default export, so you could call it whatever you'd like!

``` js
// ...
import withClasses from 'react-csjs';

@withClasses(styles)
class MyComponent extends Component { }
```

I sort of see where you're going now with the name though, but I'm torn, they both make sense depending on how you look at it:
- _withClasses()_ because it provides a `classes` prop
- _withStyles()_ because it takes in the csjs `styles` param

Maybe @wKovacs64 has some thoughts here?

**Comment by wKovacs64** [comment]
Like you said, @tizmagik, the consumer has the ability to call the HOC whatever they like and both make sense depending on interpretation. I do this, personally:

``` js
import newComponentWithInjectedClassesPropContainingCsjsStyleClassNames from 'react-csjs';
```

:wink:

As for the `classes` prop...
- I like the current name because (like you said) it is an object of class names (not raw styles) and it's easy to rename if you prefer `styles` or otherwise.
- Calling it `styles` would introduce [shadowing](http://eslint.org/docs/rules/no-shadow) when storing the CSJS result as `styles` (which appears to be the convention) if it happens to be in scope with the component definition (a common scenario). Not a big deal to fix, just don't call the CSJS results `styles`, but now you're back to deviating from common naming conventions which seems to be what sparked this discussion in the first place.
- Calling it `styles` would probably ease the migration process for consumers coming from `css-modules` (where I came from), but don't most modern IDEs have refactoring capabilities? Rename your `styles` prop to `classes` and you're done?

---

**TL;DR**
I like it the way it is, but I don't care if it changes because it's trivial to manipulate downstream.

**Comment by jonaskello** [comment]
@tizmagik @wKovacs64 Thanks for the thoughts and background on the naming! So I think for the most part it is a matter of taste. However the shadowing issue would force two different names whatever they may be. So in that case I guess `style` and `classes` is as good as any other two names :-).
